src/com/codersbay/Main.java

```diff
@@ -1,9 +1,37 @@
 package com.codersbay;
 
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
 public class Main {
 
     public static void main(String[] args) {
 
-    }
+        int[] numbers = {25, 10, 15, 4, 7, 13, 2};
+
+        for (int j = 1; j < numbers.length; j++) {
+            if (numbers[j] < numbers[j - 1]) {
+                int temp = numbers[j];
+                numbers[j] = numbers[j - 1];
+                numbers[j - 1] = temp;
+                System.out.println(Arrays.toString(numbers));
 
+                for (int i = j - 1; i >= 1; i--) {
```

Comment:
Das ist jetzt eher ein bubble sort und kein insertion sort - der Unterschied liegt darin, dass du nicht immer zwei aufeinanderfolgende elemente tauscht sondern in dem bereits sortierten teil die elemente so shiftest, dass an der stelle, an der das neue element eingefügt werden sollte ein platz "frei" wird und dann das Element an diese stelle gesetzt wird. 

Also wenn mein array ist: 
[0, 1, 4, 5, 3, 2] und ich annehme [0, 1, 4, 5] ist bereits sortiert (also mein i = 3) dann merke ich mir zuerst welches Element das nächst unsortierte ist also `nextUnsorted = 3` und dann shifte ich den sortierten Bereich solange das größte elment im sortierten array (wenn ich es von hinten durchgehe also bei 5 beginnend) größer ist als mein neues Element. 
Zwischen ergebnisse wären:  [0, 1, 4, 5, 5, 2], dann [0, 1, 4, 4, 5, 2].  
danach wird an die passende stelle das gemerkte nächste Element( also 3) eingefügt - [0, 1, 3, 4, 5, 2]